controllers/state_of_the_world.go

```diff
@@ -67,17 +104,30 @@ func NewPolicyMachineryController(manager ctrlruntime.Manager, client *dynamic.D
 			kuadrantv1beta1.LinkKuadrantToGatewayClasses,
 			kuadrantv1beta1.LinkKuadrantToLimitador,
 		),
-		controller.WithReconcile(buildReconciler(client)),
+		controller.WithReconcile(buildReconciler(client, manager.GetRESTMapper())),
 	}
 
 	ok, err := kuadrantgatewayapi.IsGatewayAPIInstalled(manager.GetRESTMapper())
 	if err != nil || !ok {
 		logger.Info("gateway api is not installed, skipping watches and reconcilers", "err", err)
 	} else {
 		controllerOpts = append(controllerOpts,
-			controller.WithRunnable("gatewayclass watcher", controller.Watch(&gwapiv1.GatewayClass{}, controller.GatewayClassesResource, metav1.NamespaceAll)),
-			controller.WithRunnable("gateway watcher", controller.Watch(&gwapiv1.Gateway{}, controller.GatewaysResource, metav1.NamespaceAll)),
-			controller.WithRunnable("httproute watcher", controller.Watch(&gwapiv1.HTTPRoute{}, controller.HTTPRoutesResource, metav1.NamespaceAll)),
+			controller.WithRunnable("gatewayclass watcher", controller.Watch(
+				&gwapiv1.GatewayClass{},
+				controller.GatewayClassesResource,
+				metav1.NamespaceAll,
+				controller.WithPredicates(&ctrlruntimepredicate.TypedGenerationChangedPredicate[*gwapiv1.GatewayClass]{})),
```

Comment:
I'm afraid GatewayClasses can change status to `Accepted: false` and `Unknown` too.

Reply:
Oh cool, I'll remove this predicate 👍